### PR TITLE
Add Transform based GPU tests for project module.

### PR DESCRIPTION
**Pull request by 1chandu**
For #2272 

#### Background
Add `Transform` based unit tests for `project` module. Verified on AMD and INTEL GPUs.

#### Change List
- Add Transform based GPU tests for project module.

### CODE REVIEW

File: test/modules/core/shaderlib/project/project-glsl.spec.js [hunk 1]
```diff
@@ -62,6 +64,67 @@ const TEST_VIEWPORT_ORTHO = new Viewport({
...
+  project_position: pos => `\
+varying vec4 outValue;
+
+void main()
+{
+  outValue = project_position(vec4(${pos[0].toFixed(MAX_FRACTION_DIGITS)}, ${pos[1].toFixed(
+    MAX_FRACTION_DIGITS
+  )}, ${pos[2].toFixed(MAX_FRACTION_DIGITS)}, ${pos[3].toFixed(
+    MAX_FRACTION_DIGITS
+  )}), vec2(0., 0.));
```
Comment (Pessimistress) [inline test/modules/core/shaderlib/project/project-glsl.spec.js]: Take advantage of the xy64Low argument to improve precision

File: test/modules/core/shaderlib/project/project-glsl.spec.js [hunk 2]
```diff
@@ -258,14 +362,17 @@ test('project#vs', t => {
     const module = projectVS(uniforms);
 
     testCase.tests.forEach(c => {
-      const actual = c.func(module);
+      let actual = c.func(module);
```
Comment (Pessimistress) [inline test/modules/core/shaderlib/project/project-glsl.spec.js]: Only run this if webgl is not available?

File: test/modules/core/shaderlib/project/project-glsl.spec.js (hunk 1 above)
Comment (1chandu) [reply test/modules/core/shaderlib/project/project-glsl.spec.js]: Added code to use low64 bits, but LNGLAT mode test still fails, as these not used in base project module. We can remove the tolerance for project-64 module tests.

File: test/modules/core/shaderlib/project/project-glsl.spec.js (hunk 2 above)
Comment (1chandu) [reply test/modules/core/shaderlib/project/project-glsl.spec.js]: Done, run tests using Transform when supported, otherwise fallback to transpiler.

**Pull request merged**